stock_move_backdating/tests/test_stock_move_backdating.py

```diff
@@ -1,106 +1,108 @@
+        date_backdating_2 = self._get_date_backdating(2)
+        self._transfer_picking_with_dates(date_backdating_1, date_backdating_2)
+
+    def _move_factory(self, product, qty):
+        return self.env['stock.move'].create({
+            'name': '/',
+            'product_id': product.id,
+            'product_uom_qty': qty,
+            'product_uom': product.uom_id.id,
+            'location_id': self.env.ref('stock.stock_location_stock').id,
+            'location_dest_id':
+            self.env.ref('stock.stock_location_customers').id,
+        })
+
+    def _transfer_details_factory(self, picking):
+        return self.env['stock.transfer_details'].with_context(
             active_model='stock.picking',
-            active_ids=[pick.id],
-            active_id=len([pick.id]) and pick.id or False
-        ).create(
-            {'picking_id': pick.id}
-        )
-        pick_wizard.item_ids[0].write({'date': back_date})
-        pick_wizard.do_detailed_transfer()
+            active_ids=[picking.id],
+            active_id=picking.id).create({'picking_id': picking.id})
 
-    def computeMove(self, timedelta_days):
+    def _get_date_backdating(self, timedelta_days):
         now = datetime.now()
-        back_date = (now - timedelta(days=timedelta_days)).strftime(
+        date_backdating = (now - timedelta(days=timedelta_days)).strftime(
             DEFAULT_SERVER_DATE_FORMAT)
-        self.create_assigne_picking()
-        self.run_picking(back_date)
-        self.picking_out.action_done()
-        move = self.move_id
-        move.refresh()
-        self.assertEqual(move.date[0:10], back_date)
-        self.assertEqual(self.picking_out.state, 'done')
-        self.assertEqual(move.state, 'done')
-        for quant in move.quant_ids:
-            self.assertEqual(quant.in_date[0:10], back_date)
-
-    def test_0_stock_move_backdate_yesterday(self):
-        self.computeMove(1)
-
-    def test_1_stock_move_backdate_last_month(self):
-        self.computeMove(31)
+        return date_backdating
 
-    def test_2_stock_move_backdate_future(self):
-        self.assertRaises(
-            orm.except_orm,
-            self.computeMove, -1)
+    def setUp(self):
+        super(TestStockMoveBackdating, self).setUp()
+        product_8 = self.env.ref('product.product_product_8')
+        product_9 = self.env.ref('product.product_product_9')
+        self.picking = self.env['stock.picking'].create({
+            'picking_type_id': self.env.ref('stock.picking_type_out').id})
+        self.picking.move_lines = (
+            self._move_factory(product=product_8, qty=1.0) |
+            self._move_factory(product=product_9, qty=2.0))
+        self.picking.action_confirm()
+        self.picking.force_assign()
+
+    def _transfer_picking_with_date(self, date_backdating):
+        wizard = self._transfer_details_factory(self.picking)
+        wizard.date_backdating = date_backdating
+        wizard.onchange_date_backdating()
```

Comment:
@eLBati, come conferma, ho aggiunto i due assert seguenti, e anche in questo caso i test automatici vengono superati.